### ISSUE TITLE
SublimeText 4 native jsx syntax does not work with SublimeLinter-eslint

### ISSUE THREAD

**Issue by KidkArolis** [SublimeLinter/SublimeLinter-eslint#310]
I've added the following log lines to the run method in linter.py:

```py
    def run(self, cmd, code):
        logger.info('  -> run(self, cmd, code):')
        logger.info(cmd)
        logger.info(code)
        # Workaround eslint bug https://github.com/eslint/eslint/issues/9515
        # Fixed in eslint 4.10.0
        if code == '':
            code = ' '

        self.ensure_plugin_installed()
        return super().run(cmd, code)
```

Below is the output of the SublimeLinter-eslint after saving a file in JSX syntax mode. What you'll notice is that

* there are 7 eslint tasks on the queue
* each `linter.py:56` log line, which is meant to be the code - is just a small slice of the code (e.g. `children`, `300`, etc.)

```
SublimeLinter: linter.py:1090         eslint: Checking lint mode 'background' vs lint reason 'on_modified'.  Ok.
SublimeLinter: sublime_linter.py:345  Delay linting 'linting.js' for 0.0s
SublimeLinter: linter.py:1090         eslint: Checking lint mode 'background' vs lint reason 'on_save'.  Ok.
SublimeLinter: backend.py:84          WARNING: 'linting.js' puts 7 eslint tasks on the queue.
SublimeLinter: backend.py:91          WARNING: 'linting.js' puts in total 7(!) tasks on the queue.
SublimeLinter: #36 linter.py:1107     eslint: linting 'linting.js'
SublimeLinter: #37 linter.py:1107     eslint: linting 'linting.js'
SublimeLinter: #38 linter.py:1107     eslint: linting 'linting.js'
SublimeLinter: #39 linter.py:1107     eslint: linting 'linting.js'
SublimeLinter: #36 node_linter.py:85  Searching executable for 'eslint' starting at '/Users/karolis/Desktop/testing'.
SublimeLinter: #37 node_linter.py:85  Searching executable for 'eslint' starting at '/Users/karolis/Desktop/testing'.
SublimeLinter: #38 node_linter.py:85  Searching executable for 'eslint' starting at '/Users/karolis/Desktop/testing'.
SublimeLinter: #39 node_linter.py:85  Searching executable for 'eslint' starting at '/Users/karolis/Desktop/testing'.
SublimeLinter: #36 linter.py:54         -> run(self, cmd, code):
SublimeLinter: #38 linter.py:54         -> run(self, cmd, code):
SublimeLinter: #39 linter.py:54         -> run(self, cmd, code):
SublimeLinter: #37 linter.py:54         -> run(self, cmd, code):
SublimeLinter: #37 linter.py:55       ['/Users/karolis/Desktop/testing/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/Users/karolis/Desktop/testing/linting.js']
SublimeLinter: #38 linter.py:55       ['/Users/karolis/Desktop/testing/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/Users/karolis/Desktop/testing/linting.js']
SublimeLinter: #39 linter.py:55       ['/Users/karolis/Desktop/testing/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/Users/karolis/Desktop/testing/linting.js']
SublimeLinter: #37 linter.py:56       styles.Spinner
SublimeLinter: #36 linter.py:55       ['/Users/karolis/Desktop/testing/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/Users/karolis/Desktop/testing/linting.js']
SublimeLinter: #38 linter.py:56       300
SublimeLinter: #39 linter.py:56       children
SublimeLinter: #36 linter.py:56       children
SublimeLinter: #40 linter.py:1107     eslint: linting 'linting.js'
SublimeLinter: #41 linter.py:1107     eslint: linting 'linting.js'
SublimeLinter: #42 linter.py:1107     eslint: linting 'linting.js'
SublimeLinter: #40 node_linter.py:85  Searching executable for 'eslint' starting at '/Users/karolis/Desktop/testing'.
SublimeLinter: #41 node_linter.py:85  Searching executable for 'eslint' starting at '/Users/karolis/Desktop/testing'.
SublimeLinter: #42 node_linter.py:85  Searching executable for 'eslint' starting at '/Users/karolis/Desktop/testing'.
SublimeLinter: #40 linter.py:54         -> run(self, cmd, code):
SublimeLinter: #41 linter.py:54         -> run(self, cmd, code):
SublimeLinter: #42 linter.py:54         -> run(self, cmd, code):
SublimeLinter: #40 linter.py:55       ['/Users/karolis/Desktop/testing/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/Users/karolis/Desktop/testing/linting.js']
SublimeLinter: #41 linter.py:55       ['/Users/karolis/Desktop/testing/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/Users/karolis/Desktop/testing/linting.js']
SublimeLinter: #42 linter.py:55       ['/Users/karolis/Desktop/testing/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/Users/karolis/Desktop/testing/linting.js']
SublimeLinter: #40 linter.py:56       toastMessages || []
SublimeLinter: #41 linter.py:56       null
SublimeLinter: #42 linter.py:56       showDemoToast && (
          <DemoToast
            hidden={showingProfileSummary}
            ready={user.isWelcomed}
            shorter={isMobile}
          />
        )
SublimeLinter: sublime_linter.py:593  Linting 'linting.js' took 0.03s
```

For comparison, here's the output of the same file but set to JavaScript syntax mode:

```
SublimeLinter: sublime_linter.py:345  Delay linting 'linting.js' for 0.1s
SublimeLinter: linter.py:1090         eslint: Checking lint mode 'background' vs lint reason 'on_modified'.  Ok.
SublimeLinter: #43 linter.py:1107     eslint: linting 'linting.js'
SublimeLinter: #43 node_linter.py:85  Searching executable for 'eslint' starting at '/Users/karolis/Desktop/testing'.
SublimeLinter: #43 linter.py:54       -> run(self, cmd, code):
SublimeLinter: #43 linter.py:55       ['/Users/karolis/Desktop/testing/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/Users/karolis/Desktop/testing/linting.js']
SublimeLinter: #43 linter.py:56       <<<REDACTED 145 LINES OF THE FULL CODE>>>

SublimeLinter: #43 linter.py:1623     Running ...

  /Users/karolis/Desktop/testing  (working dir)
  $ cat linting.js | /Users/karolis/Desktop/testing/node_modules/.bin/eslint --format json --stdin --stdin-filename /Users/karolis/Desktop/testing/linting.js

SublimeLinter: #43 linter.py:1189     eslint: output:
  [{"filePath":"/Users/karolis/Desktop/testing/linting.js" <<<REDACTED>>> }]
SublimeLinter: #43 linter.py:139      eslint output:
[{'errorCount': 2,
  'filePath': '/Users/karolis/Desktop/testing/linting.js',
  'fixableErrorCount': 0,
  'fixableWarningCount': 0,
  'messages': [{'column': 48,
                'endColumn': 56,
                'endLine': 15,
                'line': 15,
                'message': "'children' is missing in props validation",
                'messageId': 'missingPropType',
                'nodeType': 'Property',
                'ruleId': 'react/prop-types',
                'severity': 2},
               {'column': 16,
                'endColumn': 24,
                'endLine': 25,
                'line': 25,
                'message': "'children' is missing in props validation",
                'messageId': 'missingPropType',
                'nodeType': 'Property',
                'ruleId': 'react/prop-types',
                'severity': 2}],
  'source': <<<REDACTED>>>",
  'usedDeprecatedRules': [],
  'warningCount': 0}]
SublimeLinter: sublime_linter.py:593  Linting 'linting.js' took 0.47s
```

* only 1 lint process is run
* full code is passed to the linter

For what it's worth, I think `SublimeLinter-contrib-standard` works correctly, but I'm not 100% certain, need to re-check.

Any pointers as to what could be going on here with this newly launched JSX syntax and the interaction with either `SublimeLinter`, `NodeLinter` or `SublimeLinter-eslint`? Any points to help fix this would be appreciated.

For anyone else running into this issue - the workaround for now is to coninue using the Babel Sublime package to get JSX highlighting and linting working.

**Comment by kaste** [comment]
So you're running 7 eslint tasks in 0.03s.  😲  I don't think my computer can do that.  

I'm not on ST4 yet, but it comes down to 🤞 

https://github.com/SublimeLinter/SublimeLinter-eslint/blob/50210cadb9297e5aca66f87b51dd98fbbdeee4c9/linter.py#L27

changing that to `STANDARD_SELECTOR = 'source.js, source.jsx'` should do it as ST4 comes with JSX (under the name 'source.jsx') out of the box.  You can also check the selector `ctrl+alt+shift+p` in the source file, usually while the cursor is on the first char, will bring up a popup.

**Comment by orinoco** [comment]
Based on your comments, I added this to my SublimeLinter.sublime-settings

```
"linters": {
    "eslint": {
      "selector": "source.js, source.jsx"
    }
  }
```

And linting started working again in ST4!

**Comment by kaste** [comment]
Fix released as 4.2.2
